storage/s3.go

```diff
@@ -294,33 +294,9 @@ func (s *S3) listObjects(ctx context.Context, url *url.URL) <-chan *Object {
 	return objCh
 }
 
-// setAcl sets value of acl flag pointer according to the metadata
-// provided.
-func setAcl(acl **string, metadata map[string]string) error {
-	if acl == nil {
-		return fmt.Errorf("acl reference cannot point to nil")
-	}
-	aclVal := metadata["acl"]
-	if aclVal == "" {
-		return nil
-	}
-
-	supportedActions := map[string]bool{
-		"private": true, "public-read": true,
-		"public-read-write": true, "authenticated-read": true,
-		"aws-exec-read": true, "bucket-owner-read": true,
-		"bucket-owner-full-control": true, "log-delivery-write": true,
-	}
-	if _, ok := supportedActions[aclVal]; !ok {
-		return fmt.Errorf("provided acl flag value is not supported")
-	}
-	*acl = aws.String(aclVal)
-	return nil
-}
-
 // Copy is a single-object copy operation which copies objects to S3
 // destination from another S3 source.
-func (s *S3) Copy(ctx context.Context, from, to *url.URL, metadata map[string]string) error {
+func (s *S3) Copy(ctx context.Context, from, to *url.URL, _metadata map[string]string) error {
```

Comment:
no underscores :)

Reply:
what do you think of changing the signature of Copy and Put?

 ```
Copy(ctx context.Context, src, dst *url.URL, metadata Metadata) error
Put(ctx context.Context, src io.Reader, dst *url.URL, metadata Metadata, concurrency int, partSize int64) error
```

I don't see any reason to use `map[string]string` rather than `Metadata`

Reply:
I thought about doing that, but we will have to do modifications at Storage interface and other places which use map to pass metadata, and also map might be easier to reason about. Will make your suggested modification, seems logical. thanks